### ISSUE TITLE
Make "Remove" popup dialog accessible.

### ISSUE THREAD

**Issue by yzen** [mozilla-mobile/mozilla-vpn-client#138]
A couple of *remaining* things need to happen with the pop-up dialog to be delightfully accessible:
- <del>Focus handling:<del>
  - <del>on dialog opening, focus must be shifted onto the dialog itself or the first focusable control inside the dialog<del>
  - <del>focus must be contained within the dialog - only allow focus to shift within focusable controls inside the dialog (focus can either be wrapped or not)<del>
  - <del>everything "under" the dialog overlay must be disabled - setting `enabled` property to false should help with making controls not focusable (this can help with the point above)<del>
- Semantics
  - We now have `Accessible.role: Accessible.Dialog` and `Accessible.name: popupTitle.text` set on the `VPNRemoveDevicePopup` however when testing (at least on Mac) the semantics is not set on the dialog when using a screen reader. The content of the popup seems to land directly as children of the top level Window of the VPN app.
  - <del>The dialog must have dialog semantics (e.g. `Accessible.role: Accessible.Dialog`) and it must have `Accessible.name` set as well (accessible name in this case should just refer to the "Remove device?" main label).<del>
  - <del>"Please confirm..." should have `Accessible.name` to match the text or `Accessible.role: Accessible.StaticText` so it is in the screen reader virtual cursor order.<del>
  - We no longer use opacity for the popup and instead you built in `open` method of the `Popup` type, however (and probably) because Dialog semantics are not present, we still do not get the appropriate accessibility events. <del>I suspect that we should rather use visibility instead of opacity on the component that is going to have the dialog role because it would then properly notify screen readers about dialog open/close events. I doubt that works with opacity.<del>I
- <del>Keyboard<del>
  - <del>Ensure that `Escape` key closes dialog.<del>

**Comment by bakulf** [comment]
Can we consider this bug completed, or is there something more to do?

**Comment by yzen** [comment]
I think there's still an issue with Dialog semantics missing though I do not think we should close it. @bakulf perhaps removing `UX` and priority labels (unless if we have to keep priority at `P3`) would be reasonable?

**Comment by yzen** [comment]
I updated the description with the remaining list.

**Comment by yzen** [comment]
Closing in favour of #322
